src/Adapter/LinearRegressionAdapter.php

```diff
@@ -2,6 +2,102 @@
 
 namespace Zeeml\Algorithms\Adapter;
 
+use Zeeml\Algorithms\Dataset\DatasetInterface;
+
 class LinearRegressionAdapter extends AbstractAdapter
 {
+    protected $intercept;
+    protected $slope;
+
+    /**
+     * LinearRegressionAdapter constructor.
+     * @param DatasetInterface $dataset
+     */
+    public function __construct(array $data)
```

Comment:
I prefer an object rather an array...
We could have a simple toArray() method then.
An object could be changed from outside the instance with a trigger to force preparation again in classification/prediction adapter... just a thought.